flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/polardbx/PolardbxSourceTCase.java

```diff
@@ -0,0 +1,472 @@
+                                + ") WITH ("
+                                + " 'connector' = 'mysql-cdc',"
+                                + " 'scan.incremental.snapshot.enabled' = 'true',"
+                                + " 'hostname' = '%s',"
+                                + " 'port' = '%s',"
+                                + " 'username' = '%s',"
+                                + " 'password' = '%s',"
+                                + " 'database-name' = '%s',"
+                                + " 'table-name' = '%s',"
+                                + " 'scan.incremental.snapshot.chunk.size' = '100',"
+                                + " 'server-id' = '%s'"
+                                + ")",
+                        HOST_NAME,
+                        PORT,
+                        USER_NAME,
+                        PASSWORD,
+                        DATABASE,
+                        getTableNameRegex(captureCustomerTables),
+                        getServerId());
+
+        // first step: check the snapshot data
+        String[] snapshotForSingleTable =
+                new String[] {
+                    "+I[4, 1002, 2, 106, 2022-01-16T00:00]",
+                    "+I[5, 1003, 1, 107, 2022-01-16T00:00]",
+                    "+I[2, 1002, 2, 105, 2022-01-16T00:00]",
+                    "+I[3, 1004, 3, 109, 2022-01-16T00:00]",
+                    "+I[1, 1001, 1, 102, 2022-01-16T00:00]",
+                };
+        tEnv.executeSql(sourceDDL);
+        TableResult tableResult = tEnv.executeSql("select * from orders_with_multi_pks");
+        CloseableIterator<Row> iterator = tableResult.collect();
+        List<String> expectedSnapshotData = new ArrayList<>();
+        for (int i = 0; i < captureCustomerTables.length; i++) {
+            expectedSnapshotData.addAll(Arrays.asList(snapshotForSingleTable));
+        }
+
+        List<String> realSnapshotData = fetchRows(iterator, expectedSnapshotData.size());
+        assertEqualsInAnyOrder(expectedSnapshotData, realSnapshotData);
+
+        // second step: check the sink data
+        tEnv.executeSql(
+                "CREATE TABLE multi_key_sink ("
+                        + " id BIGINT NOT NULL,"
+                        + " seller_id STRING,"
+                        + " order_id STRING,"
+                        + " buyer_id STRING,"
+                        + " create_time TIMESTAMP,"
+                        + " primary key (id,order_id) not enforced"
+                        + ") WITH ("
+                        + " 'connector' = 'values',"
+                        + " 'sink-insert-only' = 'false'"
+                        + ")");
+        TableResult sinkResult =
+                tEnv.executeSql("insert into multi_key_sink select * from orders_with_multi_pks");
+
+        waitForSinkSize("multi_key_sink", realSnapshotData.size());
+        assertEqualsInAnyOrder(
+                expectedSnapshotData, TestValuesTableFactory.getRawResults("multi_key_sink"));
```

Comment:
In above three cases we can validate binlog reading as well just like other ConnectorITCase,  we can make sure the polardbx works well under snapshot phase and binlog phase in this way

Reply:
thanks for suggestion, I have added some binlog reading cases, it works